util/src/kvdb.rs

```diff
@@ -417,11 +418,10 @@ impl Database {
 		};
 	}
 
-	/// Commit buffered changes to database.
-	pub fn flush(&self) -> Result<(), String> {
+	/// Commit buffered changes to database. Must be called under `flush_lock`
```

Comment:
yup - even for internal functions, it's pretty easy for someone else to come along and call it. docs all too easily can get out of date. functions that require locks to act properly are inherently dangerous. a minimum would be to append `_with_lock` to the function name.